During startup: "java.sql.SQLException: statement is not executing"

Tested on a clean server. The plugin appears to load but doesn't actually function. /cprivate prints the 'left-click a block' message, but it doesn't respond to the left-click action.

Issue reproduced on commit edd2724. Could not reproduce on commit c2bb8dd8.

[11:06:01] [Server thread/INFO]: [LWC] Loading LWC v${project.version}
[11:06:01] [Server thread/INFO]: [LWC] Enabling LWC v${project.version}
[11:06:01] [Server thread/INFO]: [LWC] Connecting to SQLite
[11:06:01] [Server thread/WARN]: com.griefcraft.scripting.ModuleException: java.sql.SQLException: statement is not executing
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.sql.Database.printException(Database.java:204)
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.resolveProtections(PhysDB.java:814)
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdate400_5(PhysDB.java:2036)
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:271)
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.lwc.LWC.load(LWC.java:1540)
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:203)
[11:06:01] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
[11:06:01] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337)
[11:06:01] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:402)
[11:06:01] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:384)
[11:06:01] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:333)
[11:06:01] [Server thread/WARN]: 	at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:206)
[11:06:01] [Server thread/WARN]: 	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:545)
[11:06:01] [Server thread/WARN]: 	at java.lang.Thread.run(Unknown Source)
[11:06:01] [Server thread/WARN]: Caused by: java.sql.SQLException: statement is not executing
[11:06:01] [Server thread/WARN]: 	at org.sqlite.core.CoreStatement.checkOpen(CoreStatement.java:50)
[11:06:01] [Server thread/WARN]: 	at org.sqlite.jdbc3.JDBC3PreparedStatement.executeQuery(JDBC3PreparedStatement.java:65)
[11:06:01] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.resolveProtections(PhysDB.java:811)
[11:06:01] [Server thread/WARN]: 	... 12 more
[11:06:01] [Server thread/ERROR]: Error occurred while enabling LWC v${project.version} (Is it up to date?)
com.griefcraft.scripting.ModuleException: java.sql.SQLException: statement is not executing
	at com.griefcraft.sql.Database.printException(Database.java:204) ~[?:?]
	at com.griefcraft.sql.PhysDB.precache(PhysDB.java:873) ~[?:?]
	at com.griefcraft.lwc.LWC.load(LWC.java:1549) ~[?:?]
	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:203) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337) [spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:402) [spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:384) [spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:333) [spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:206) [spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:545) [spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_144]
Caused by: java.sql.SQLException: statement is not executing
	at org.sqlite.core.CoreStatement.checkOpen(CoreStatement.java:50) ~[spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at org.sqlite.core.CorePreparedStatement.batch(CorePreparedStatement.java:117) ~[spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at org.sqlite.jdbc3.JDBC3PreparedStatement.setInt(JDBC3PreparedStatement.java:324) ~[spigot-1.12.2.jar:git-Spigot-5695bca-53fccdf]
	at com.griefcraft.sql.PhysDB.precache(PhysDB.java:862) ~[?:?]
	... 10 more

--- a/core/src/main/java/com/griefcraft/sql/Database.java
+++ b/core/src/main/java/com/griefcraft/sql/Database.java
@@ -339,7 +339,8 @@ public abstract class Database {
         try {
             if (useStatementCache) {
                 Statistics.addQuery();
-                return statementCache.get(sql, () -> prepareInternal(sql, returnGeneratedKeys));
+                final PreparedStatement p = statementCache.getIfPresent(sql);
+				if(p != null) return p;
             }
 
             return prepareInternal(sql, returnGeneratedKeys);

